sdk/cosmos/azure-cosmos-spark-fabric-account-data-resolver/src/main/scala/com/azure/cosmos/spark/fabric/FabricAccountDataResolver.scala

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos.spark.fabric
+
+import com.azure.cosmos.spark.AccountDataResolver
+import com.azure.cosmos.spark.CosmosAccessToken
+import com.azure.cosmos.spark.CosmosConfigNames.AccountDataResolverServiceName
+import com.azure.cosmos.spark.CosmosConstants.Names.getFabricAccountDataResolverFQDN
+import com.fasterxml.jackson.databind.ObjectMapper
+
+import java.time.{Instant, OffsetDateTime, ZoneOffset}
+import scala.collection.concurrent.TrieMap
+
+class FabricAccountDataResolver extends AccountDataResolver with BasicLoggingTrait {
+
+  override def getAccountDataConfig(configs: Map[String, String]): Map[String, String] = {
+    val configParameters = ConfigParameters.apply(configs)
+    if (isEnabled(configParameters)) {
+      configs + ("spark.cosmos.auth.type" -> "AccessToken")
+    } else {
+      configs
+    }
+  }
+
+  private def isEnabled(configs: ConfigParameters): Boolean = {
+    configs.accountDataResolverServiceName.isDefined && configs.accountDataResolverServiceName.get
+      .equalsIgnoreCase(getFabricAccountDataResolverFQDN)
+  }
+
+  override def getAccessTokenProvider(configs: Map[String, String]): Option[List[String] => CosmosAccessToken] = {
+    val configParameters = ConfigParameters.apply(configs)
+    FabricAccountDataResolver.tokenProviders.getOrElseUpdate(
+      configParameters,
+      getAccessTokenProviderImpl(configParameters)
+    )
+  }
+
+  private def getAccessTokenProviderImpl(configs: ConfigParameters): Option[List[String] => CosmosAccessToken] = {
+    if (isEnabled(configs)) {
+      logInfo(s"FabricAccountDataResolver is enabled")
+      Some((_: List[String]) => {
+        // obtains the access token from fabric environment
+        val accessToken = mssparkutils.credentials.getToken(configs.audience.getOrElse(FabricAccountDataResolver.AUDIENCE))
```

Comment:
QQ: how long the token will be valid for? and have we manually validated at least for one round that token works and refresh fine?